python_modules/dagster/dagster/_cli/dev.py

```diff
@@ -326,3 +346,116 @@ def _workspace_opts_to_serialized_cli_args(workspace_opts: WorkspaceOpts) -> Seq
         args.append("--use-ssl")
 
     return args
+
+
+class DagsterDevCodeServerManager:
```

Comment:
this can subclass AbstractContextManager i think since it has enter and exit methods https://docs.python.org/3/library/contextlib.html#contextlib.AbstractContextManager

Reply:
great point.